packages/picasso.js/src/web/components/tooltip/placement.js

```diff
@@ -0,0 +1,416 @@
+}
+
+function alignToPoint({
+  options,
+  pointer,
+  width,
+  height,
+  dockOrder,
+  x,
+  y
+}) {
+  const {
+    targetBounds
+  } = pointer;
+
+  // Check if explicit dock
+  const dockTransforms = getDockTransform(options.offset);
+  const transform = dockTransforms[options.dock];
+  if (transform) {
+    return {
+      computedTooltipStyle: {
+        left: `${x}px`,
+        top: `${y}px`,
+        transform
+      },
+      computedArrowStyle: getComputedArrowStyle(options.offset)[options.dock],
+      dock: options.dock
+    };
+  }
+
+  const area = {
+    width: options.area === 'target' ? targetBounds.width : window.innerWidth,
+    height: options.area === 'target' ? targetBounds.height : window.innerHeight
+  };
+  const dockOffsets = getDockOffset(width, height, options.offset);
+
+  const results = [];
+  const edgeMargin = 20;
+  const vx = options.area === 'target' ? x : targetBounds.left + x;
+  const vy = options.area === 'target' ? y : targetBounds.top + y;
+
+  for (let i = 0; i < dockOrder.length; i += 1) {
+    const dock = dockOrder[i];
+
+    const offset = calcOffset({
+      area, vx, vy, width, height, offset: dockOffsets[dock]
+    });
+
+    const computedTooltipStyle = {
+      left: `${x}px`,
+      top: `${y}px`,
+      transform: dockTransforms[dock]
+    };
+    const computedArrowStyle = getComputedArrowStyle(options.offset)[dock];
+
+    if (offset.x !== 0) {
+      computedTooltipStyle.width = `${width - edgeMargin - Math.abs(offset.x)}px`;
+
+      if (dock === 'top' || dock === 'bottom') {
+        computedTooltipStyle.left = `${x + offset.x}px`;
+
+        computedArrowStyle.left = `calc(50% ${offset.x > 0 ? '-' : '+'} ${Math.abs(offset.x)}px)`;
+      }
+    }
+
+    const result = {
+      computedTooltipStyle,
+      computedArrowStyle,
+      dock,
+      rect: {
+        width,
+        height
+      }
+    };
+
+    if (offset.x === 0 && offset.y === 0) {
+      return result;
+    }
+
+    result.offset = offset;
+
+    results.push(result);
+  }
+
+  results.sort((a, b) => Math.abs(a.offset.x) - Math.abs(b.offset.x));
+
+  return results[0];
+}
+
+function alignToPointer({
+  options,
+  pointer,
+  width,
+  height
+}) {
+  const {
+    x,
+    y
+  } = pointer;
+
+  return alignToPoint({
+    x,
+    y,
+    pointer,
+    width,
+    height,
+    options,
+    dockOrder: ['top', 'left', 'right', 'bottom']
+  });
+}
+
+function alignToSlice({
+  options,
+  pointer,
+  width,
+  height,
+  nodes,
+  resources
+}) {
+  const node = nodes[0];
+  const { dx, dy } = pointer;
+  const componentBounds = resources.getComponentBoundsFromNode(node);
+
+  // cx and cy relative to targetBounds
+  const center = {
+    x: dx + componentBounds.x + (componentBounds.width / 2),
+    y: dy + componentBounds.y + (componentBounds.height / 2)
+  };
+
+  const {
+    start,
+    end,
+    outerRadius
+  } = node.desc.slice;
+
+  // origin is at 12 o clock, clockwise, so it is transform to origin at 3 a clock
```

Comment:
Could you add a comment that it starts counting clockwise at Pi / 2 and that these calculations start before the transform? Would be greatly appreciated for me :)

Reply:
Well it sort of does state that. By I'll try and make it more clear